mentat/code_file.py

```diff
@@ -53,6 +84,87 @@ def __init__(self, path: str | Path):
                 self.intervals = [Interval(0, math.inf)]
             else:
                 self.intervals = parse_intervals(split[1])
+                level = CodeMessageLevel.INTERVAL
+        self.level = level
+        self.diff = diff
+
+    def __repr__(self):
+        return (
+            f"CodeFile(fname={self.path.name}, intervals={self.intervals},"
+            f" level={self.level}, diff={self.diff})"
+        )
 
     def contains_line(self, line_number: int):
         return any([interval.contains(line_number) for interval in self.intervals])
+
+    async def _get_file_message(
+        self,
+        config: ConfigManager,
+        code_file_manager: "CodeFileManager",
+        parser: "Parser",
+    ) -> list[str]:
+        file_message: list[str] = []
+
+        # We always want to give GPT posix paths
+        abs_path = Path(config.git_root / self.path)
+        rel_path = Path(os.path.relpath(abs_path, config.git_root))
+        posix_rel_path = Path(rel_path).as_posix()
+        file_message.append(posix_rel_path)
+
+        if self.level in {CodeMessageLevel.CODE, CodeMessageLevel.INTERVAL}:
+            file_lines = code_file_manager.read_file(abs_path)
+            for i, line in enumerate(file_lines, start=1):
+                if self.contains_line(i):
+                    if parser.provide_line_numbers():
+                        file_message.append(f"{i}:{line}")
+                    else:
+                        file_message.append(f"{line}")
+        elif self.level == CodeMessageLevel.CMAP_FULL:
+            cmap = await get_code_map(config.git_root, self.path)
+            file_message += cmap
+        elif self.level == CodeMessageLevel.CMAP:
+            cmap = await get_code_map(
+                config.git_root, self.path, exclude_signatures=True
+            )
+            file_message += cmap
+        elif not CodeMessageLevel.FILE_NAME:
+            raise ValueError(f"Invalid code message level: {self.level}")
+        file_message.append("")
+
+        if self.diff is not None:
+            diff: str = get_diff_for_file(config.git_root, self.diff, rel_path)
+            diff_annotations = parse_diff(diff)
+            if self.level == CodeMessageLevel.CODE:
+                file_message = annotate_file_message(file_message, diff_annotations)
+            else:
+                for section in diff_annotations:
+                    file_message += section.message
+        return file_message
+
+    _file_checksum: str | None = None
+    _file_message: list[str] | None = None
+
+    async def get_code_message(
```

Comment:
I don't like the naming `get_code_message` and `_get_file_message` because it makes it sound like code_message and file_message are different things. Maybe `get_code_message` and `_create_code_message`. Or honestly just `get_code_message` and `_get_code_message` because then it's clear they return the same value but one is for internal use only. Also for consistency I'd have the classes internal variable be `_code_message` not `_file_message`.